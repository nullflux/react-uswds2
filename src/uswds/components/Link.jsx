import React from "react";
import { NavLink as RRLink } from "react-router-dom";

/**
 * Represents a generic link element.
 *
 * This class is a wrapper around the React Router Link class and accepts any 
 * props supported by that class.
 *
 * If no `activeClassName` is defined, this class will set it to the default 
 * value of "usa-current" per the U.S. Web Design Standards.
 */
export default class Link extends React.Component
{
  /**
   * Renders the component.
   *
   * @returns {Node|null} The rendered DOM node.
   */
  render()
  {
    let props = Object.assign({}, this.props);
    if (!props.activeClassName) {
      props.activeClassName = "usa-current";
    }
    return (
      <RRLink {...props} />
    );
  }
}
