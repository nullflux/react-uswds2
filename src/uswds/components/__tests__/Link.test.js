import React from 'react';
import { shallow } from 'enzyme';
import Link from '../Link.jsx';

jest.unmock('../Link.jsx');

describe('Link', () => {
  it('should render correctly', () => {
    const dest = '/testURL';
    const component = shallow(
    <Link to={dest}>This is a link</Link>
    );
    expect(component).toMatchSnapshot();
  });
});

