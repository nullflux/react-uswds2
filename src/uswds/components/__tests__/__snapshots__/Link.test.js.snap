// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<NavLink
  activeClassName="usa-current"
  to="/testURL"
>
  This is a link
</NavLink>
`;
